audinterface/core/utils.py

```diff
@@ -138,6 +154,90 @@ def segments_to_indices(
     return starts_i, ends_i
 
 
+def signal_index(
+        starts: Timestamps = None,
+        ends: Timestamps = None,
+) -> pd.MultiIndex:
+    r"""Create signal index.
+
+    Args:
+        starts: segment start positions.
+            Time values given as float or integers are treated as seconds
+        ends: segment end positions.
+            Time values given as float or integers are treated as seconds
+
+    Returns:
+        signal index
+
+    Raises:
+        ValueError: if created index contains duplicates
+
+    Raises:
+        ValueError: if ``start`` and ``ends`` differ in size
```

Comment:
```suggestion
    Raises:
        ValueError: if created index contains duplicates
        ValueError: if ``start`` and ``ends`` differ in size
```

Reply:
Actually, we don't check for duplicates, so I removed it. Since we do not deal with annotations here, I don't we have to check for duplicates.

Reply:
You can get `pandas` errors for duplicated index entries, but it depends what you are doing. The advantage of duplicates is that we can force an object to process the same segment more than one time.

Reply:
But there is one downside, further down you set `signal_index` is also a `segmented_index` only without the file level. This would then be no longer true, as a segmented index is not allowed to contain duplicates.

Reply:
So should we raise an error for duplicates then?

Reply:
I'm still not completely sure.
Maybe it is handy to allow for duplicated index entries.

The advantage of not allowing duplicates is that you only have to add the file level and you have a segmented index.

Any thoughts what would be more important?

Reply:
Maybe it's better to forbid duplicates and later allow them if we see the necessity, as the other way round would be a breaking change?

Reply:
Good argument, then let's first forbid duplicates.

Reply:
Mhh just realized that we have the following code in `Feature`:

```python
        if n_time_steps > 1:
            if win_dur is None:
                starts = [start] * n_time_steps
                ends = [end] * n_time_steps
```

so basically if we have a process function that returns more than one frame, but `win_dur` is set to `None` we duplicate `start` and `end` for each frame. This would not be possible any longer if we do not allow for duplicates. On the other hand, this will also create a bad `audformat`  index when applied on files I think....

Reply:
Yes, sounds we shouldn't doing this.
I would say hat we should not allow for `win_dur` `None` in this case, or we have to return only one time step and ignore the `n_time_steps` setting.

Reply:
Yes, we indeed get an error as the following minimal example shows:

```python
import audinterface


feature = audinterface.Feature(
    ['f1', 'f2'],
    process_func=lambda x, sr: np.ones((1, 2, 3))  # return 3 frames
)

index = audformat.segmented_index('test.wav')

feature.process_index(index)
```
```
ValueError: Index not conform to audformat. Found duplicates:
('test.wav', Timedelta('0 days 00:00:00'), Timedelta('0 days 00:00:00.000187500'))
('test.wav', Timedelta('0 days 00:00:00'), Timedelta('0 days 00:00:00.000187500'))
```

With `process_signal_from_index()` this still works and that's why do not get an error in our tests at the moment.

Reply:
> I would say hat we should not allow for win_dur None in this case, or we have to return only one time step and ignore the n_time_steps setting.

Ok, this will be a breaking change, thought I doubt we have code that will break so far. So should I continue in this PR or do it in a separate one?

Reply:
It's a breaking change, but the current behavior is simply a bug.

I will merge the current stage and propose to fix the bug in another pull request and also forbid duplicates for `signal_index()` in another pull request.